lightblue/src/main/java/org/esbtools/eventhandler/lightblue/LockedResource.java

```diff
@@ -0,0 +1,37 @@
+/*
+ *  Copyright 2016 esbtools Contributors and/or its affiliates.
+ *
+ *  This file is part of esbtools.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package org.esbtools.eventhandler.lightblue;
+
+import java.io.Closeable;
+
+public interface LockedResource extends Closeable {
```

Comment:
Semantics are a little different (throws with custom exception message instead of return false), plus I don't want to rely on lightblue client, in the client it's a class not an interface, and also we may have other impls in the future.